src/components/PageCard.jsx

```diff
@@ -0,0 +1,148 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import axios from 'axios';
+import PropTypes from 'prop-types';
+import { Base64 } from 'js-base64';
+import {
+  prettifyPageFileName, frontMatterParser, concatFrontMatterMdBody, generatePageFileName,
+} from '../utils';
+
+export default class PageCard extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      sha: '',
+      title: '',
+      permalink: '',
+      settingsIsActive: false,
+      mdBody: null,
+    };
+  }
+
+  async componentDidMount() {
+    try {
+      const { siteName, fileName } = this.props;
+      const resp = await axios.get(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/pages/${fileName}`, {
+        withCredentials: true,
+      });
+      const { sha, content } = resp.data;
+
+      // split the markdown into front matter and content
+      const { frontMatter, mdBody } = frontMatterParser(Base64.decode(content));
+      const { title, permalink } = frontMatter;
+      this.setState({
+        sha, title, permalink, mdBody,
+      });
+    } catch (err) {
+      console.log(err);
+    }
+  }
+
+  saveHandler = async () => {
+    try {
+      const { siteName, fileName } = this.props;
+      const {
+        sha, title, permalink, mdBody,
+      } = this.state;
+
+      const frontMatter = { title, permalink };
+
+      // here, we need to re-add the front matter of the markdown file
+      const upload = concatFrontMatterMdBody(frontMatter, mdBody);
+
+      // encode to Base64 for github
+      const base64Content = Base64.encode(upload);
+      const newFileName = generatePageFileName(title);
```

Comment:
does not add a `.md` as an extension to the filename.
Currently causes parsing issues when it was used on the demo-v2 site.